difffile.h

```diff
@@ -66,6 +66,14 @@ int add_RR(namedb_type* db, const dname_type* dname,
 	buffer_type* packet, size_t rdatalen, zone_type *zone,
 	int* softfail);
 
+#define CALLED_FROM_SERVER_PROCESS 0
+#define CALLED_FROM_XFRD_PROCESS   1
+
+/* apply the xfr file identified by xfrfilenr to zone */
+int apply_ixfr_for_zone(struct nsd* nsd, zone_type* zone, FILE* in,
+        struct nsd_options* opt, udb_base* taskudb, udb_ptr* last_task,
+        uint32_t xfrfilenr, uint8_t called_from);
```

Comment:
I don't really like this, especially the `called_from` argument. Before xfrd did not have any knowlegde about `zone_type` I believe. I expect this to pull in a bit much for what we need it to do(?)

Reply:
I agree this is a bit ugly yes. The reason for `called_from` is that is may not exit() when called from xfrd which is just fine and the way to fail a reload when called from main. I think I can change this by letting the function always return, and doing the exit() based on return code from main.

Reply:
Resolved in 8b93cc99